[BUG] cannot get file after user removal

## Overview
>Po usunięciu usera z listy userów store'a przestaje działać pobieranie (storeApi.openFile) wcześniej istniejących plików (jak dodam pliki po usunięciu usera to te nowe pliki działają). Wersje: 2.2.3 i 2.2.6. openFile wywala w konsoli:
endpoint-wasm-module.js:8 Error on em_crypto.methodCaller call from C for aes256CbcHmac256Decrypt {data: Int8Array(160), key: Int8Array(32), taglen: 16}
endpoint-wasm-module.js:8 Error: Wrong message security tag

--- a/endpoint/store/include_pub/privmx/endpoint/store/StoreException.hpp
+++ b/endpoint/store/include_pub/privmx/endpoint/store/StoreException.hpp
@@ -69,6 +69,7 @@ DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, FileVersionMismatchException,
 DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, StorePublicDataMismatchException, "Store public data mismatch", 0x0020)
 DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, FilePublicDataMismatchException, "File public data mismatch", 0x0021)
 DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, WritingToFileInteruptedWrittenDataSmallerThenDeclaredException, "Writing to file interupted. Written data smaller then declared", 0x0022)
+DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, FileDecryptionFailedException, "FileDecryptionFailed", 0x0022)
 
 } // store
 } // endpoint

--- a/endpoint/store/include_pub/privmx/endpoint/store/StoreException.hpp
+++ b/endpoint/store/include_pub/privmx/endpoint/store/StoreException.hpp
@@ -69,7 +69,7 @@ DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, FileVersionMismatchException,
 DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, StorePublicDataMismatchException, "Store public data mismatch", 0x0020)
 DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, FilePublicDataMismatchException, "File public data mismatch", 0x0021)
 DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, WritingToFileInteruptedWrittenDataSmallerThenDeclaredException, "Writing to file interupted. Written data smaller then declared", 0x0022)
-DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, FileDecryptionFailedException, "FileDecryptionFailed", 0x0022)
+DECLARE_ENDPOINT_EXCEPTION(EndpointStoreException, FileDecryptionFailedException, "FileDecryptionFailed", 0x0023)
 
 } // store
 } // endpoint

--- a/endpoint/store/src/StoreApiImpl.cpp
+++ b/endpoint/store/src/StoreApiImpl.cpp
@@ -366,7 +366,7 @@ int64_t StoreApiImpl::openFile(const std::string& fileId) {
     auto storeFileGetModel = utils::TypedObjectFactory::createNewObject<server::StoreFileGetModel>();
     storeFileGetModel.fileId(fileId);
     auto file_raw = _serverApi->storeFileGet(storeFileGetModel);
-    auto key = _keyProvider->getKey(file_raw.store().keys(), file_raw.store().keyId());
+    auto key = _keyProvider->getKey(file_raw.store().keys(), file_raw.file().keyId());
     auto decryptionParams = getStoreFileDecryptionParams(file_raw.file(), key);
     return createFileReadHandle(decryptionParams);
 }
@@ -376,6 +376,9 @@ StoreApiImpl::StoreFileDecryptionParams StoreApiImpl::getStoreFileDecryptionPara
         // When meta is not string, then is new V4 format as object
         auto encryptedFileMeta = utils::TypedObjectFactory::createObjectFromVar<server::EncryptedFileMetaV4>(file.meta());
         auto fileMeta = _fileMetaEncryptorV4.decrypt(encryptedFileMeta, encKey.key);
+        if(fileMeta.statusCode != 0) {
+            throw FileDecryptionFailedException("file decryption Failed with status code: " + std::to_string(fileMeta.statusCode));
+        }
         auto internalMeta = utils::TypedObjectFactory::createObjectFromVar<dynamic::InternalStoreFileMeta>(utils::Utils::parseJson(fileMeta.internalMeta.stdString()));
         if ((uint64_t)internalMeta.chunkSize() > SIZE_MAX) {
             throw NumberToBigForCPUArchitectureException("chunkSize to big for this CPU architecture");

--- a/endpoint/tests_e2e/tests/StoreTest.cpp
+++ b/endpoint/tests_e2e/tests/StoreTest.cpp
@@ -2104,3 +2104,56 @@ TEST_F(StoreTest, listStores_query) {
         }
     }
 }
+
+TEST_F(StoreTest, updateStore_key_update_open_file) {
+    std::string fileId;
+    EXPECT_NO_THROW({
+        storeApi->updateStore(
+            reader->getString("Store_1.storeId"),
+            std::vector<core::UserWithPubKey>{
+                core::UserWithPubKey{
+                    .userId=reader->getString("Login.user_1_id"),
+                    .pubKey=reader->getString("Login.user_1_pubKey")
+                },
+                core::UserWithPubKey{
+                    .userId=reader->getString("Login.user_2_id"),
+                    .pubKey=reader->getString("Login.user_2_pubKey")
+                }
+            },
+            std::vector<core::UserWithPubKey>{
+                core::UserWithPubKey{
+                    .userId=reader->getString("Login.user_1_id"),
+                    .pubKey=reader->getString("Login.user_1_pubKey")
+                },
+                core::UserWithPubKey{
+                    .userId=reader->getString("Login.user_2_id"),
+                    .pubKey=reader->getString("Login.user_2_pubKey")
+                }
+            },
+            core::Buffer::from("public"),
+            core::Buffer::from("private"),
+            1,
+            true,
+            true
+        );
+    });
+    int64_t read_handle = 0;
+    std::string read_data;
+    EXPECT_NO_THROW({
+        read_handle = storeApi->openFile(
+            reader->getString("File_1.info_fileId")
+        );
+    });
+    EXPECT_NO_THROW({
+        read_data = storeApi->readFromFile(
+            read_handle,
+            reader->getInt64("File_1.size")
+        ).stdString();
+    });
+    EXPECT_NO_THROW({
+        storeApi->closeFile(
+            read_handle
+        );
+    });
+       
+}

